### PR TITLE
Health Checking: Fix doc errors and expand

### LINKED ISSUE #1229: Health Checking Documentation bug - doesn't restart before Ready
> But base on the Health Failure Strategy
> `1. If any of the GameServer container fails exits before the GameServer moves to Ready then, it is restart as per the restartPolicy (which defaults to “Always”)`
> I didn't change the restartPolicy and gameserver describe events part didnt print `SDK.Ready() complete` why the game server didn't restart and only the gameserver in fleet will restart?

Ah, that is a legitimate documentation bug! Nice find! :+1: 

It should read: `If any of the GameServer container exits with an error before the GameServer moves to Ready then, the game server container is restarted as per the restartPolicy (which defaults to “Always”)`

And it should also state `if the GameServer health check fails at any point, the GameServer is moved to Unhealthy`. We should also have a section in here on what happens with Fleets in GameServers when they move to Unhealthy -- which is they get _deleted_ (not restarted, like above), and then recreated. Which is the difference you are seeing.
 
_Originally posted by @markmandel in https://github.com/googleforgames/agones/issues/1217#issuecomment-564357986_

/assign @markmandel

### PR DESCRIPTION
**Pull request by markmandel**
There was an erroneous statement in the documentation on how health checking and process exit would impact if a GameServer would become Unhealthy. This commit fixes this issue.

This also adds a section on how Unhealthy GameServers are handled when in a Fleet vs when there are individually created.

Closes #1229

### CODE REVIEW

**Comment by roberthbailey** [comment]
The preview site doesn't seem to be showing your changes (https://e351e4b-dot-preview-dot-agones-images.appspot.com/site/docs/guides/health-checking/). Not sure why.

File: site/content/en/docs/Guides/health-checking.md [hunk 1]
```diff
@@ -10,30 +10,40 @@ description: >
 
 ## Disabling Health Checking
 
-By default, health checking is enabled, but it can be turned off by setting the `health > disabled` property to true
+By default, health checking is enabled, but it can be turned off by setting the `spec.health.disabled` property to true
```
Comment (roberthbailey) [inline site/content/en/docs/Guides/health-checking.md]: Nit: end this sentence with a period.

File: site/content/en/docs/Guides/health-checking.md [hunk 2]
```diff
@@ -10,30 +10,40 @@ description: >
...
 The following is the process for what happens to a `GameServer` when it is unhealthy.
 
-1. If any of the GameServer container fails health checking, or exits before the GameServer moves to `Ready` then, 
-   it is restart as per the `restartPolicy` (which defaults to "Always")
-1. If the GameServer container fails healthy checking after the `Ready` state, then it doesn't restart, 
-   but moves the GameServer to an `Unhealthy` state.
-1. If the GameServer container exits while in `Ready` state, it will be restarted as per the `restartPolicy` 
-   (which defaults to "Always", since `RestartPolicy` is a Pod wide setting), 
+1. If the `GameServer` container exits with an error before the `GameServer` moves to `Ready` then, 
+   it is restart as per the `restartPolicy` (which defaults to "Always").
```
Comment (roberthbailey) [inline site/content/en/docs/Guides/health-checking.md]: nit: s/restart/restarted

File: site/content/en/docs/Guides/health-checking.md [hunk 3]
```diff
@@ -10,30 +10,40 @@ description: >
...
+   but moves to an `Unhealthy` state.
+1. If the `GameServer` container exits while in `Ready`, `Allocated` or `Reserved` state, it will be restarted 
+   as per the `restartPolicy`  (which defaults to "Always", since `RestartPolicy` is a `Pod` wide setting), 
    but will immediately move to an `Unhealthy` state.
 1. If the SDK sidecar fails, then it will be restarted, assuming the `RestartPolicy` is Always/OnFailure.
 
-## Reference
+## Fleet Management of Unhealthy GameServers
+
+If a `GameServer` moves into an `Unhealthy` state when it is not part of a Fleet, the `GameServer` will remain in the
```
Comment (roberthbailey) [inline site/content/en/docs/Guides/health-checking.md]: "remain in that state" or "remain in the unhealthy state"

File: site/content/en/docs/Guides/health-checking.md [hunk 4]
```diff
@@ -10,30 +10,40 @@ description: >
...
+until explicitly deleted.  This is useful for debugging `Unhealthy` `GameServers`, or if you are creating
+your own `GameServer` management layer, you can explicitly choose what to do if a `GameServer` becomes `Unehalthy`.
```
Comment (roberthbailey) [inline site/content/en/docs/Guides/health-checking.md]: Unhealthy is misspelled.

**Comment by markmandel** [comment]
> The preview site doesn't seem to be showing your changes (https://e351e4b-dot-preview-dot-agones-images.appspot.com/site/docs/guides/health-checking/). Not sure why.

The demos site only retains the last 30 builds, so 7 days is plenty of time for that to go away.

**Pull request merged**